buildkitd/Earthfile

```
@@ -58,8 +58,12 @@ buildkitd:
     ENTRYPOINT ["/usr/bin/entrypoint.sh", "buildkitd", "--config=/etc/buildkitd.toml"]
     ARG EARTHLY_TARGET_TAG_DOCKER
     ARG TAG="dev-$EARTHLY_TARGET_TAG_DOCKER"
-    ARG DOCKERHUB_USER="earthly"
-    SAVE IMAGE --push --cache-from=earthly/buildkitd:main $DOCKERHUB_USER/buildkitd:$TAG
+    ARG REGISTRY_USER="earthly"
+    ARG IMAGE_REGISTRY="docker.io"
+
+    # Ensure that the image we save is a fully qualified name. Podman defaults to "localhost", while Docker defaults to
+    # "docker.io". This keeps it consistent across both frontends.
+    SAVE IMAGE --push --cache-from=earthly/buildkitd:main $IMAGE_REGISTRY/$REGISTRY_USER/buildkitd:$TAG
```

do we also need to update the cache-from?
```suggestion
    SAVE IMAGE --push --cache-from=$IMAGE_REGISTRY/earthly/buildkitd:main $IMAGE_REGISTRY/$REGISTRY_USER/buildkitd:$TAG
```

---

Probably? I haven't hit any errors without it. Wouldn't hurt at least.